src/main/kotlin/org/rust/ide/typing/assist/RsSmartEnterProcessor.kt

```diff
@@ -36,6 +39,10 @@ class RsSmartEnterProcessor : SmartEnterProcessorWithFixers() {
             val elementType = each.node.elementType
             when {
                 elementType == LBRACE || elementType == RBRACE -> continue@loop
+                each is RsMatchArm && each.parent is RsMatchBody -> {
```

Comment:
`each.parent is RsMatchBody` it's always true according to our current grammar so I think we can omit it